src/cargo/util/toml/mod.rs

```diff
@@ -1922,7 +1959,21 @@ impl TomlManifest {
                     .clone()
                     .resolve_with_self(n, |dep| dep.resolve(n, inheritable, cx))?;
                 let dep = resolved.to_dependency(n, cx, kind)?;
-                validate_package_name(dep.name_in_toml().as_str(), "dependency name", "")?;
+                let name_in_toml = dep.name_in_toml().as_str();
+                validate_package_name(name_in_toml, "dependency name", "")?;
+                let kind_name = &kind
+                    .map(|k| match k {
+                        DepKind::Normal => "dependencies",
+                        DepKind::Development => "dev-dependencies",
+                        DepKind::Build => "build-dependencies",
+                    })
```

Comment:
Since this is duplicated code with `DepTable::kind_table`, could that instead be moved to a method of `DepKind` and reuse it?